cpg-core/src/main/java/de/fraunhofer/aisec/cpg/graph/declarations/FunctionDeclaration.kt

```diff
@@ -0,0 +1,333 @@
+        for (functionDeclaration in c) {
+            addOverriddenBy(functionDeclaration)
+        }
+    }
+
+    fun addOverriddenBy(functionDeclaration: FunctionDeclaration) {
+        addIfNotContains(overriddenBy, functionDeclaration)
+    }
+
+    fun addOverrides(functionDeclaration: FunctionDeclaration) {
+        addIfNotContains(overrides, functionDeclaration)
+    }
+
+    fun addThrowTypes(type: Type) {
+        throwsTypes.add(type)
+    }
+
+    fun addThrowTypes(collection: Collection<Type>) {
+        for (type in collection) {
+            addThrowTypes(type)
+        }
+    }
+
+    fun <T> getBodyStatementAs(i: Int, clazz: Class<T>): T? {
+        if (body is CompoundStatement) {
+            val statement = (body as CompoundStatement?)!!.statements[i] ?: return null
+            return if (clazz.isAssignableFrom(statement.javaClass)) clazz.cast(statement) else null
+        }
+        return null
+    }
+
+    /**
+     * A list of default expressions for each item in [parameters]. If a [ParamVariableDeclaration]
+     * has no default, the list will be null at this index. This list must have the same size as
+     * [parameters].
+     */
+    val defaultParameters: List<Expression?>
+        get() {
+            return parameters.map { it.default }
+        }
+
+    val defaultParameterSignature: List<Type>
+        get() {
+            val signature: MutableList<Type> = ArrayList()
+            for (paramVariableDeclaration in parameters) {
+                if (paramVariableDeclaration.default != null) {
+                    signature.add(paramVariableDeclaration.getType())
+                } else {
+                    signature.add(UnknownType.getUnknownType())
+                }
+            }
+            return signature
+        }
+
+    val signatureTypes: List<Type>
+        get() {
+            val signatureTypes: MutableList<Type> = ArrayList()
+            for (paramVariableDeclaration in parameters) {
+                signatureTypes.add(paramVariableDeclaration.getType())
+            }
+            return signatureTypes
+        }
+
+    fun addParameter(paramVariableDeclaration: ParamVariableDeclaration) {
+        val propertyEdge = PropertyEdge(this, paramVariableDeclaration)
+        propertyEdge.addProperty(Properties.INDEX, parameters.size)
+        parameterEdges.add(propertyEdge)
+    }
+
+    fun removeParameter(paramVariableDeclaration: ParamVariableDeclaration) {
+        parameterEdges.removeIf { propertyEdge: PropertyEdge<ParamVariableDeclaration> ->
+            propertyEdge.end == paramVariableDeclaration
+        }
+    }
+
+    /**
+     * Looks for a variable declaration by the given name.
+     *
+     * @param name the name of the variable
+     * @return an optional value containing the variable declaration if found
+     */
+    fun getVariableDeclarationByName(name: String?): Optional<VariableDeclaration> {
+        return if (body is CompoundStatement) {
```

Comment:
This can be significantly compressed

Reply:
I didn't even see that this function exists, should be replaced with our new extension functions.